openstack/baremetal/v1/ports/requests.go

```diff
@@ -0,0 +1,223 @@
+package ports
+
+import (
+  "fmt"
+
+  "github.com/gophercloud/gophercloud"
+  "github.com/gophercloud/gophercloud/pagination"
+)
+
+// ListOptsBuilder allows extensions to add additional parameters to the
+// List request.
+type ListOptsBuilder interface {
+	ToPortListQuery() (string, error)
+  ToPortListDetailQuery() (string, error)
+}
+
+// ListOpts allows the filtering and sorting of paginated collections through
+// the API. Filtering is achieved by passing in struct field values that map to
+// the node attributes you want to see returned. Marker and Limit are used
+// for pagination.
+type ListOpts struct{
+  // Filter the list by the name or uuid of the Node
+  Node string `q:"node"`
+
+  // Filter the list by the Node uuid
+  NodeUUID string `q:"node_uuid"`
```

Comment:
This is unfortunate, node_uuid is so ancient (API 1.6, Liberty era?), I'd love to avoid it. I wonder if we can default to microversion 1.6 and get rid of this..

Reply:
Is it not supported on 1.6 or later? It's still listed in the most current API docs

Reply:
Well,  was in the documentation .-.

Reply:
I think Dmitry might be thinking of /v1/nodes/$uuid/ports/, although that is not how the API is documented.. and I think we should conform to how the API is documented.